fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+		return
+	message_admins("[key_name_admin(user)] has removed [name] antagonist status from [key_name_admin(owner)].")
+	log_admin("[key_name(user)] has removed [name] antagonist status from [key_name(owner)].")
+	on_removal()
+
+// Buying powers
+/datum/antagonist/bloodsucker/proc/BuyPower(datum/action/bloodsucker/power)//(obj/effect/proc_holder/spell/power)
+	powers += power
+	power.Grant(owner.current) // owner.AddSpell(power)
+
+/datum/antagonist/bloodsucker/proc/AssignStarterPowersAndStats()
+	update_hud(owner.current) 	// Set blood value, current rank
+	// Powers
+	BuyPower(new /datum/action/bloodsucker/feed)
+	BuyPower(new /datum/action/bloodsucker/masquerade)
+	BuyPower(new /datum/action/bloodsucker/veil)
+	// Traits
+	if(HAS_TRAIT(owner.current, TRAIT_TOXINLOVER)) //No slime bonuses here, no thank you
+		had_toxlover = TRUE
+		REMOVE_TRAIT(owner.current, TRAIT_TOXINLOVER, SPECIES_TRAIT)
+	for(var/T in defaultTraits)
+		ADD_TRAIT(owner.current, T, BLOODSUCKER_TRAIT)
+	// Traits: Species
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		var/datum/species/S = H.dna.species
+		S.species_traits |= DRINKSBLOOD
+	// Clear Addictions
+	owner.current.reagents.addiction_list = list() // Start over from scratch. Lucky you! At least you're not addicted to blood anymore (if you were)
+	// Stats
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		var/datum/species/S = H.dna.species
+		// Make Changes
+		H.physiology.brute_mod *= 0.8
+		H.physiology.cold_mod = 0
+		H.physiology.stun_mod *= 0.5
+		H.physiology.siemens_coeff *= 0.75 	//base electrocution coefficient  1
+		S.punchdamagelow += 1       //lowest possible punch damage   0
+		S.punchdamagehigh += 1      //highest possible punch damage	 9
+		if(istype(H) && owner.assigned_role == "Clown")
+			H.dna.remove_mutation(CLOWNMUT)
+			to_chat(H, "As a vampiric clown, you are no longer a danger to yourself. Your nature is subdued.")
+	// Physiology
+	CheckVampOrgans() // Heart, Eyes
+	// Language
+	owner.current.grant_language(/datum/language/vampiric)
+	// Disabilities
+	CureDisabilities()
+
+/datum/antagonist/bloodsucker/proc/ClearAllPowersAndStats()
+	// Powers
+	while(powers.len)
+		var/datum/action/bloodsucker/power = pick(powers)
+		powers -= power
+		power.Remove(owner.current)
+		// owner.RemoveSpell(power)
+	// Traits
+	if(had_toxlover)
+		ADD_TRAIT(owner.current, TRAIT_TOXINLOVER, SPECIES_TRAIT)
+
+	// Traits: Species
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		H.set_species(H.dna.species.type)
+	// Stats
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		H.set_species(H.dna.species.type)
+		// Clown
+		if(istype(H) && owner.assigned_role == "Clown")
+			H.dna.add_mutation(CLOWNMUT)
+	for(var/T in defaultTraits)
+		REMOVE_TRAIT(owner.current, T, BLOODSUCKER_TRAIT)
+	// Physiology
+	owner.current.regenerate_organs()
+	// Update Health
+	owner.current.setMaxHealth(100)
```

Comment:
I'd recommend using initial() here instead of a flat value of 100

also, something something brave bull, heretic ghouls, voiceless dead, etc. shapeshifting might also get weird.